analysis/src/CreateInterface.ml

```diff
@@ -162,15 +162,39 @@ let printSignature ~extractor ~signature =
           Buffer.add_string buf "\n"
       in
       processSignature ~indent rest
-    | Sig_value (_id, {val_kind = Val_prim prim; val_loc}) :: items
+    | Sig_value (id, ({val_kind = Val_prim prim; val_loc} as vd)) :: items
       when prim.prim_native_name <> "" && prim.prim_native_name.[0] = '\132' ->
-      (* Rescript primitive name, e.g. @val external ...
-         Copy the external declaration verbatim from the implementation file *)
+      (* Rescript primitive name, e.g. @val external ... *)
       let lines =
         let posStart, posEnd = Loc.range val_loc in
         extractor |> SourceFileExtractor.extract ~posStart ~posEnd
       in
-      Buffer.add_string buf ((lines |> String.concat "\n") ^ "\n");
+      let firstLine =
+        match lines with
+        | [] -> ""
+        | firstLine :: _ -> firstLine
+      in
+      let inlineAttr = "@inline" in
+      let inlineAttrLen = String.length inlineAttr in
+      let attrStartIdx = String.index firstLine '@' in
+
+      if
+        try String.sub firstLine attrStartIdx inlineAttrLen = inlineAttr
+        with _ -> false
```

Comment:
One would catch the specific exception `| Invalid_argument _ ->` .

But here one can also check the size beforehand.